src/overlays/actors/ovl_En_Kbt/z_en_kbt.c

```diff
@@ -250,7 +250,7 @@ void func_80B34314(EnKbt* this, PlayState* play) {
     } else if (this->actor.xzDistToPlayer < 250.0f) {
         if ((this->unk_278 != NULL) && (this->unk_278->xzDistToPlayer < 250.0f)) {
             if (this->unk_27C & 4) {
-                playerForm = gSaveContext.save.playerForm;
+                playerForm = GET_PLAYER_FORM;
                 if (((playerForm ^ 0) != PLAYER_FORM_HUMAN) || ((CUR_FORM_EQUIP(EQUIP_SLOT_B) != ITEM_SWORD_KOKIRI) &&
```

Comment:
Is this temp still required with the macro?

Reply:
```c
void func_80B34314(EnKbt* this, PlayState* play) {
    func_80B34078(this);

    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
        this->actionFunc = func_80B34598;
        Actor_ChangeFocus(&this->actor, play, this->unk_278);
        this->unk_278->textId = this->actor.textId;
        this->unk_27C &= ~4;
        if (this->actor.textId == 0xC4E) {
            this->unk_27C |= 0x10;
        }
    } else if (this->actor.xzDistToPlayer < 250.0f) {
        if ((this->unk_278 != NULL) && (this->unk_278->xzDistToPlayer < 250.0f)) {
            if (this->unk_27C & 4) {
                if ((GET_PLAYER_FORM != PLAYER_FORM_HUMAN) || ((CUR_FORM_EQUIP(EQUIP_SLOT_B) != ITEM_SWORD_KOKIRI) &&
                                                                (CUR_FORM_EQUIP(EQUIP_SLOT_B) != ITEM_SWORD_RAZOR) &&
                                                                (CUR_FORM_EQUIP(EQUIP_SLOT_B) != ITEM_SWORD_GILDED))) {
                    this->actor.textId = 0xC38;
                } else if (CURRENT_DAY == 3) {
                    this->actor.textId = 0xC39;
                } else if (this->unk_27C & 8) {
                    this->actor.textId = 0xC3E;
                } else {
                    this->actor.textId = 0xC3A;
                }
            }

            switch(this->actor.textId) {
                case 0xC4E:
                case 0xC4F:
                case 0xC50:
                    if (GET_PLAYER_FORM != PLAYER_FORM_HUMAN) {
                        this->actor.textId = 0xC37;
                    }
                    break;

                case 0xC37:
                    if (GET_PLAYER_FORM == PLAYER_FORM_HUMAN) {
                        if (func_80B33E8C(play)) {
                            this->actor.textId = 0xC50;
                        } else {
                            this->actor.textId = 0xC4E;
                        }
                    }
                    break;

                default:
                    break;
            }
            func_800B8614(&this->actor, play, 260.0f);
        }
    }
    func_80B3415C(this);
}
```
seems to work

Reply:
Nice find! Need to keep an eye out for more potential switches that are hidden behind `!=`'s